src/gui/SampleWaveform.cpp

```diff
@@ -44,12 +44,12 @@ void SampleWaveform::visualize(Parameters parameters, QPainter& painter, const Q
 	const float resolution = std::max(1.0f, framesPerPixel / maxFramesPerPixel);
 	const float framesPerResolution = framesPerPixel / resolution;
 
-	const size_t numPixels = std::min<size_t>(parameters.size, width);
+	size_t numPixels = std::min(parameters.size, static_cast<size_t>(width));
 	auto min = std::vector<float>(numPixels, 1);
 	auto max = std::vector<float>(numPixels, -1);
 	auto squared = std::vector<float>(numPixels, 0);
 
-	const size_t maxFrames = numPixels * static_cast<size_t>(framesPerPixel);
+	const size_t maxFrames = static_cast<size_t>(static_cast<float>(numPixels) * framesPerPixel);
```

Comment:
The `static_cast<float>` should not be necessary here because the result of a multiplication of an `int` and `float` should be a `float` already.

Reply:
I will try that… C++ isn't my main language so I originally opted for explicit casting to make my life easier